### ISSUE TITLE
Understanding the SupCon loss

### ISSUE THREAD

**Issue by Mojzaar** [HobbitLong/SupContrast#98]
Hi,

Thank you for supplying the paper's code. I had a great time reading the paper and experimenting with the code.

I had a few concerns about the loss. When I had a normalized random vector such as X, with dimension (B,1, N) as my anchor images in a self-supervised mode where we don't have a mask and label since the anchors and augmented version are identical, I should receive 0 if I feed torch.cat((X,X),1) to the loss function, right? However, instead of 0, I get a positive value. Indeed, raising the batch size increases the loss values, and I only get zero when the batch size is one (B=1, N any other number)! Is there something I'm missing here?

I wanted to use this loss to maximize the mutual information between anchor images and different forms of that image resulting from various processes. As a result, my data loader delivers a batch of anchor images along with the corresponding noisy instances, all of which are the same batch size. However, if I use a batch size larger than one, I will not receive the correct result!

I'd be grateful if you could assist me in gaining a better knowledge of your implementation.
